### PR TITLE
issue-65-Django-Secrets

### LINKED ISSUE #65: Django secrets
Having all secrets saved in **.env** file
These should include database username and password and any other  key that isn't supposed to be visible 
`Can this be put in an env file instead of in code?

_Originally posted by @nyandika in https://github.com/SpaceyaTech/Team-Rio-Django/pull/48#discussion_r1030619058_`

### PR DESCRIPTION
**Pull request by OyugoObonyo**
# Description

This pull request handles the creation of an .env.sample file that contains the fields required in the .env file. 

Fixes #65 

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CMS/settings.py [hunk 1]
```diff
@@ -31,54 +33,57 @@
...
-    'django.contrib.auth.middleware.AuthenticationMiddleware',
-    'django.contrib.messages.middleware.MessageMiddleware',
-    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+    "django.middleware.security.SecurityMiddleware",
+    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django.middleware.common.CommonMiddleware",
+    "django.middleware.csrf.CsrfViewMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "django.contrib.messages.middleware.MessageMiddleware",
+    "django.middleware.clickjacking.XFrameOptionsMiddleware",
```
Comment (JimmyTron) [inline CMS/settings.py:56]: You went through the hustle of changing all single quotes to double quotes.. 😂😂🤧

File: CMS/settings.py [hunk 2]
```diff
@@ -88,38 +93,43 @@
...
+STATIC_URL = "static/"
 
 # Default primary key field type
 # https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field
+MEDIA_URL = "/media/"
+MEDIA_ROOT = os.path.join(BASE_DIR, "media")
+
+DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
 
-DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
```
Comment (JimmyTron) [inline CMS/settings.py:125]: What prompted removal of this field?

File: CMS/settings.py [hunk 3]
```diff
@@ -31,54 +33,57 @@
...
 
 DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': BASE_DIR / 'db.sqlite3',
+    "default": {
+        "ENGINE": "django.db.backends.sqlite3",
+        "NAME": BASE_DIR / "db.sqlite3",
     }
 }
```
Comment (JimmyTron) [inline CMS/settings.py:88]: We should have changed this to Postgresql. its long overdue

File: env.sample [hunk 4]
```diff
@@ -0,0 +1,12 @@
+-----
+This is a sample of the field required in the developer's .env file.
+The characters in capitals letters are the required environment variables.
```
Comment (JimmyTron) [inline env.sample:3]: Nice 👍👍

**Review by JimmyTron — APPROVED** [review]
It had quite a number of files in there. This should have been done on the main branch though... We'll see how to move them.

File: CMS/settings.py (hunk 1 above)
Comment (OyugoObonyo) [reply CMS/settings.py:56]: I used Black to do this. We should insist on a style guide for this project, and maybe include style checks in our CI/CD workflow as well.

File: CMS/settings.py (hunk 2 above)
Comment (OyugoObonyo) [reply CMS/settings.py:125]: I don' t recall deleting this on the settings.py file. If I did, it must have been by accident. Apologies.

File: CMS/settings.py (hunk 3 above)
Comment (OyugoObonyo) [reply CMS/settings.py:88]: Once the .env thing is settled, next up is deciding on a DB. PostgreSQL is obviously the favorite candidate.

File: requirements.txt [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
+mypy-extensions==0.4.3
+pathspec==0.10.3
+phonenumbers==8.13.2
+Pillow==9.3.0
+platformdirs==2.6.0
+python-dotenv==0.21.0
 pytz==2022.6
 sqlparse==0.4.3
+tomli==2.0.1
 tzdata==2022.6
```
Comment (JimmyTron) [inline requirements.txt:16]: you must have forgotten to add the dotenv module to the requirements file

**Review by JimmyTron — APPROVED** [review]

**Review by JimmyTron — APPROVED** [review]

File: CMS/settings.py [hunk 6]
```diff
@@ -9,10 +9,12 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/4.0/ref/settings/
 """
+from dotenv import load_dotenvll
```
Comment (JimmyTron) [inline CMS/settings.py:12]: Screen Shot 2022-12-26 at 5 23 19 PM

File: CMS/settings.py [hunk 7]
```diff
@@ -21,7 +22,7 @@
 # See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'django-insecure-0(d^jk^v&wtq5jnh4sz8a+5q_rd-w6zm(l=)n$)*2hl1mcdlqb'
+SECRET_KEY = os.environ.get("SECRET_KEY")
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
```
Comment (sangkips) [inline CMS/settings.py:29]: Can you add DEBUG = True to .env also. This is how I normally do. `DEBUG = bool(int(os.environ.get('DEBUG', '1')))
`

**Comment by sangkips** [comment]
I have seen a lot of deletions and additions in this code, this issue arises as a result of different ways of formatting, as a start I have added black to the requirement file but moving forward we can have linters such as flake8

**Review by wanjirumurira — APPROVED** [review]

**Review by JimmyTron — APPROVED** [review]

File: CMS/settings.py (hunk 2 above)
Comment (Barissa-Imran) [reply CMS/settings.py:140]: The field in question still exist line 133-140 in green

**Pull request merged**